v2/apiserver/internal/core/mongodb/events_store.go

```diff
@@ -285,7 +286,7 @@ func (e *eventsStore) CancelMany(
 				},
 			},
 		); err != nil {
-			return events, errors.Wrap(err, "error updating events")
+			return eventCh, errCh, errors.Wrap(err, "error updating events")
```

Comment:
It would probably be more clean to return nils here instead of open channels. Same in a few other error handling conditionals below.